plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -99,7 +100,19 @@ public FlightClientManager(
      * @return An OpenSearchFlightClient instance for the specified node
      */
     public Optional<FlightClient> getFlightClient(String nodeId) {
-        return Optional.ofNullable(flightClients.get(nodeId));
+        ClientHolder clientHolder = flightClients.getOrDefault(nodeId, null);
+        return clientHolder == null ? Optional.empty() : Optional.ofNullable(clientHolder.flightClient);
```

Comment:
I think the `flightClient` should never be `null`, right?
```suggestion
        return clientHolder == null ? Optional.empty() : Optional.of(clientHolder.flightClient);
```